geopandas/geodataframe.py

```diff
@@ -1408,6 +1409,8 @@ def __finalize__(self, other, method=None, **kwargs):
         # concat operation: using metadata of the first object
         elif method == "concat":
             for name in self._metadata:
+                #  if name == '_crs':
+                #  continue
```

Comment:
Thanks @gcaria for looking at this in more detail than I have had the chance to!
I've just quickly checked out your branch and I think this will do it(?) to get things to work for both axis 0 and 1
```suggestion
                if name == '_crs' and self._crs is not None:
                    continue
```
The logic being that if you fix the `axis=0` case in `array.py` then it will already be set at finalize and nothing needs to change. For axis =1, information can be lost and the crs can be None, in which case it is copied from the first gdf in concat. 

If that logic makes sense and seems right to you, it might also be worth adding a comment as to why this extra conditional case is needed.

Reply:
Thanks @m-richards ! All tests now pass with the changes you suggest, so I have committed them